_data/glossary/statefulset.yml

```diff
@@ -0,0 +1,18 @@
+id: statefulset
+name: StatefulSet
+formerly:
+- PetSet
+related:
+- Deployment
+- Pod
+tags:
+- Storage
+- Workload
+- API Object
+short-description: |
+  Manages the deployment and scaling of a set of replica Pods, *and provides guarantees about ordering*. It does so by maintaining a *unique*, sticky identity for each of its Pods.
+
+long-description: |
+  Like a Deployment, a StatefulSet manages Pods that are based on an identical container spec. However, although their specs are the same, the Pods in a StatefulSet are not interchangeable. Each Pod has a persistent identifier that it maintains across any rescheduling.
+
+  StatefulSets also operate according to the Controller pattern. You define your desired state in a StatefulSet *object*, and the StatefulSet *controller* makes any necessary updates to the get there from the current state.
```

Comment:
I am thinking of writing an additional doc (separate from glossary) to document the Controller pattern. It's widely used across Kubernetes but there isn't a good high level explanation anywhere. just fyi